### DISCUSSION TITLE
I'm not able to insert findings variables in a table

### DISCUSSION THREAD

**Discussion by OlderMutt** [Syslifters/sysreptor#318] [Q&A] [answered]
Hello! i need to create a table like this
![image](https://github.com/user-attachments/assets/23da4a63-f2fa-408e-a197-b2b975625bcc)
but i cannot find how can i populate this table automatically with the findings variables.
Do you have any documentation about all the variables ?
Thank you

**Comment by aronmolnar** [comment]
Hi,
In your report, there is an array called "findings". The finding-elements hold the variables that you can define in your "Finding Fields":

![image](https://github.com/user-attachments/assets/8224b9c0-f514-43b9-b790-f1391a2be4f7)

For example in the columns `Vulnerabilità` you can use `{{ finding.title }}`.  
If you need more or other fields, you can just define them in the "Finding Fields" section.

The "Demo Calzone" design has a similar table that you can use as a starting point:

```
<table>
    <thead>
        <tr>
            <th>#</th>
            <th>CVSS</th>
            <th>Description</th>
            <th>Page</th>
        </tr>
    </thead>
    <tbody>
        <tr v-for="finding in findings" class="table-row-link" :class="'risk-bg-' + finding.cvss.level">
            <td>
                <ref :to="finding.id">{{ ['I', 'L', 'M', 'H', 'C'][finding.cvss.level_number - 1] }}{{ findings.filter(f => f.cvss.level === finding.cvss.level).indexOf(finding) + 1 }}</ref>
            </td>
            <td>
                <ref :to="finding.id">{{ finding.cvss.score }}</ref>
            </td>
            <td>
                <ref :to="finding.id">{{ finding.title }} {{ finding.cwe.name }}</ref>
            </td>
            <td>
                <ref :to="finding.id" class="ref-page" />
            </td>
        </tr>
    </tbody>
</table>
```

![image](https://github.com/user-attachments/assets/772efdcc-2277-4414-b281-8e1ede2a1c6d)

**Reply by OlderMutt** [reply]
Thank you very much !

**Reply by OlderMutt** [reply]
is there a way to put in the table the OWASP Top 10 category (like A03 - Injection) taken from the vulnerability template ?

**Reply by aronmolnar** [reply, marked as answer]
Sure, you can add `owasp_top10_2021` as a finding field. This is an `enum` object that you can then use in your report, e.g. `{{ finding.owasp_top10_2021.label }}`. This would print "A01:2021 - Broken Access Control" to your report.

![image](https://github.com/user-attachments/assets/793fb8f1-94c2-47e1-acfb-35508e4ae100)

You can use lodash to postprocess the label, e.g. strip `:2021` by using  `lodash.replace(finding.owasp_top10_2021.label, ':2021', '')`

**Reply by OlderMutt** [reply]
Awesome! thanks

**Reply by OlderMutt** [reply]
is There  way to order the findings from Critical to Info?

**Reply by aronmolnar** [reply]
You can define your default ordering in the `Finding Fields` menu.

![image](https://github.com/user-attachments/assets/cd8f4346-c6b1-4195-9719-5f81bee5ea0d)

This is, for example, first cvss, then title in the Calzone demo report. If you use the field `severity` you can set it there.  
If you want to deviate from your standard ordering settings, you can sort your findings during your table loop using lodash:

`v-for="finding in lodash.sortBy(findings, 'severity.value')"`

or

`v-for="finding in lodash.sortBy(findings, 'severity.value').reverse()"` for descending order.
